docs/book/src/topics/managedcluster.md

```diff
@@ -610,6 +610,111 @@ spec:
     enablePrivateClusterPublicFQDN: false # Allowed only when enablePrivateCluster is true
 ```
 
+### OS configurations of Linux agent nodes (AKS)
+
+Reference:
+
+- [How-to-guide Linux OS Custom Configuration](https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration#linux-os-custom-configuration)
+- [AKS API definition Linux OS Config](https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration#linux-os-custom-configuration)
+
+When you create your node pool (`AzureManagedMachinePool`), you can specify configuration which tunes the linux OS configuration on all nodes in that pool. For example:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureManagedMachinePool
+metadata:
+  name: "${CLUSTER_NAME}-pool1"
+spec:
+  linuxOSConfig:
+    swapFileSizeMB: 1500
+    sysctls:
+      fsAioMaxNr: 65536
+      fsFileMax: 8192
+      fsInotifyMaxUserWatches: 781250
+      fsNrOpen: 8192
+      kernelThreadsMax: 20
+      netCoreNetdevMaxBacklog: 1000
+      netCoreOptmemMax: 20480
+      netCoreRmemDefault: 212992
+      netCoreRmemMax: 212992
+      netCoreSomaxconn: 163849
+      netCoreWmemDefault: 212992
+      netCoreWmemMax: 212992
+      netIpv4IPLocalPortRange:
+        first: 32000
+        last: 60000
+      netIpv4NeighDefaultGcThresh1: 128
+      netIpv4NeighDefaultGcThresh2: 512
+      netIpv4NeighDefaultGcThresh3: 1024
+      netIpv4TCPFinTimeout: 5
+      netIpv4TCPKeepaliveProbes: 1
+      netIpv4TCPKeepaliveTime: 30
+      netIpv4TCPMaxSynBacklog: 128
+      netIpv4TCPMaxTwBuckets: 8000
+      netIpv4TCPTwReuse: true
+      netIpv4TcpkeepaliveIntvl: 10
+      netNetfilterNfConntrackBuckets: 65536
+      netNetfilterNfConntrackMax: 131072
+      vmMaxMapCount: 65530
+      vmSwappiness: 10
+      vmVfsCachePressure: 15
+    transparentHugePageDefrag: "defer+madvise"
+    transparentHugePageEnabled: "madvise"
+```
+
+Below are the full set of AKS-supported `linuxOSConfig` configurations. All properties are children of the `spec.linuxOSConfig` configuration in an `AzureManagedMachinePool` resource:
+
+| Configuration               | Property Type     | Allowed Value(s)                                                                         |
+|-----------------------------|-------------------|------------------------------------------------------------------------------------------|
+| `swapFileSizeMB`            | integer           | minimum value `1`                                                                        |
```

Comment:
@CecileRobertMichon Does CAPZ publish generated API docs like CAPI: https://cluster-api.sigs.k8s.io/reference/api_reference.html? I think this kind of information would better fit somewhere like that where it's automated.

Reply:
it's published automatically at https://doc.crds.dev/github.com/kubernetes-sigs/cluster-api-provider-azure (not by us) but it tends to be a few versions behind. We could probably link to it somewhere in our book like CAPI does.